index.html

```diff
@@ -619,30 +619,45 @@ <h4>Processing considerations</h4>
+      is an [=event handler IDL attribute=] for the `onconfigurationchange`
+      [=event handler=], whose [=event handler event type=] is
+      <dfn event for=MediaStreamTrack>configurationchange</dfn>.
+    </p>
     <p>
       <p>When the [=User Agent=] detects <dfn data-export id="change-track-configuration">a change of configuration</dfn>
-       in a <var>track</var>'s underlying source, the [=User Agent=] MUST run the following steps:</p>
+       in a <var>track</var>'s underlying <var>source</var>, the [=User Agent=] MUST run the following steps:</p>
       <ol>
         <li><p>If <var>track</var>.{{MediaStreamTrack/muted}} is <code>true</code>, wait for <var>track</var>.{{MediaStreamTrack/muted}}
           to become <code>false</code> or <var>track</var>.{{MediaStreamTrack/readyState}} to be "ended".</p></li>
         <li><p>If <var>track</var>.{{MediaStreamTrack/readyState}} is "ended", abort these steps.</p></li>
-        <li><p>If <var>track</var>'s capabilities, constraints and settings are matching <var>source</var> configuration, abort these steps.
         <li>
-          <!-- FIXME: Export capabilities, constraints and settings so that we can use them here. -->
-          <p>Update <var>track</var>'s capabilities, constraints and settings according <var>track</var>'s underlying source.</p>
+          <!-- FIXME: Export capabilities and settings so that we can use them here. -->
+          <p>Update <var>track</var>'s capabilities and settings according <var>track</var>'s underlying source.</p>
         </li>
-        <li>
-          <p>[=Fire an event=] named <var>configurationchange</var> on <var>track</var>.</p>
+        <li><p>Let <var>changes</var> be an empty [=set=].</p></li>
+        <li><p>[=map/For each=] <var>key</var>-<var>value</var> pair of <var>source</var>'s settings, if <var>track</var>'s settings does not have the same pair, [=set/append=] <var>key</var> to <var>changes</var>.</p></li>
+        <li><p>[=map/For each=] <var>key</var>-<var>value</var> pair of <var>track</var>'s settings, if <var>source</var>'s settings does not have the same pair, [=set/append=] <var>key</var> to <var>changes</var>.</p></li>
+        <li><p>[=map/For each=] <var>key</var>-<var>value</var> pair of <var>source</var>'s capabilities, if <var>track</var>'s capabilities does not have the same pair, [=set/append=] <var>key</var> to <var>changes</var>.</p></li>
+        <li><p>[=map/For each=] <var>key</var>-<var>value</var> pair of <var>track</var>'s capabilities, if <var>source</var>'s capabilities does not have the same pair, [=set/append=] <var>key</var> to <var>changes</var>.</p></li>
+        <li><p>If <var>changes</var> [=list/is empty=], abort these steps.</p></li>
+        <li><p>[=list/Sort=] <var>changes</var> in lexicographical order.</p></li>
+        <li>     
+          <p>[=Fire an event=] named {{MediaStreamTrack/configurationchange}}
```

Comment:
I think we should also specify the boolean values of `bubbles` and `cancellable` for this event. I'd say they should be both `false`. Do you agree?

Reply:
These are the default values in EventInit so we might not need it.